src/routes/safe/components/Apps/communicator.ts

```diff
@@ -0,0 +1,103 @@
+import { MutableRefObject, useEffect, useState } from 'react'
+import {
+  getSDKVersion,
+  SDKMessageEvent,
+  MethodToResponse,
+  Methods,
+  ErrorResponse,
+  MessageFormatter,
+} from '@gnosis.pm/safe-apps-sdk'
+import { SafeApp } from './types.d'
+
+type MessageHandler = (
+  msg: SDKMessageEvent,
+) => void | MethodToResponse[Methods] | ErrorResponse | Promise<MethodToResponse[Methods] | ErrorResponse | void>
```

Comment:
ErrorResponse is duplicated

Reply:
In German we say "Doppelt hält besser" (https://www.dict.cc/?s=Doppelt+h%C3%A4lt+besser#:~:text=Doppelt%20h%C3%A4lt%20besser!&text=Make%20assurance%20double%20sure.,proverb%20Better%20safe%20than%20sorry.)

Reply:
But the second one is as the param of the Promise, so it is actually a different thing ;)

So it is `void | MethodToResponse[Methods] | ErrorResponse` or as Promise that returns one of these :)